### ISSUE TITLE
Training not working on custom dataset

### ISSUE THREAD

**Issue by omsrisagar** [lllyasviel/ControlNet#273]
Dear @lllyasviel ,

Thank you for your amazing work and providing the code. I am trying to train on my custom dataset consisting of pairs of building sketches and images. I followed your tutorial (train.md) under docs (started from pretrained SD v1.5 and v2.1 models and tutorial_train.py script). After several steps of training, it seems the original colors are lost and the output looks more like filling in the shapes in the sketch than realistic. This is a dataset of 80k sketch-image pairs where sketches are obtained during DexiNed and the prompt for all is "a high quality detailed professional image"

The following is the result for starting the finetuning from SD2.1 model (at 148k steps, epoch 8):

![samples_cfg_scale_9 00_gs-148943_e-000007_b-003300](https://user-images.githubusercontent.com/9084686/225128929-be5093de-1d20-407d-b051-6aea777b5143.png)
![control_gs-148943_e-000007_b-003300](https://user-images.githubusercontent.com/9084686/225128952-01429b41-cff5-4206-a3c7-520a0d5145ac.png)
![reconstruction_gs-148943_e-000007_b-003300](https://user-images.githubusercontent.com/9084686/225128984-81326259-cbb4-4686-90cf-32427255b386.png)

The same for SD1.5 model (at 44k steps, epoch 2):

![gs-044542_e-000001_b-016800](https://user-images.githubusercontent.com/9084686/225129277-a9f9490a-880d-4faa-b93d-2938e9fde954.png)

Interestingly the results are great using your scribble2image pretrained model:

![scene](https://user-images.githubusercontent.com/9084686/225127087-630559aa-132b-487e-8f57-647832cdf166.png)

1. Any suggestions on where I could be doing wrong?
2. Would it be possible to provide the link to your scribble2image training code, so I can compare and see where there is an issue?

Thank you!

**Comment by qingfengmingyue** [comment]
This happened to me too, and the loss has been oscillating without convergence

**Comment by lllyasviel** [comment]
sd training need prompts. see also blip

**Comment by qingfengmingyue** [comment]
> sd training need prompts. see also blip

I used blip to generate prompts, but the effect is still not good. I retrained in the pre-training mode（control_sd15_canny.pth） you provided, and the effect is worse than your pre-training,batch_size=4, 8 GPU ecpoch=12 ,train data num=50000

**Comment by qingfengmingyue** [comment]
![image](https://user-images.githubusercontent.com/7944239/225518258-ff0a365d-81b2-4bbc-9800-05103a233a04.png)


> > sd training need prompts. see also blip
> 
> I used blip to generate prompts, but the effect is still not good. I retrained in the pre-training mode（control_sd15_canny.pth） you provided, and the effect is worse than your pre-training, learning_rate = 1e-5 sd_locked = True only_mid_control = False strategy="ddp" precision=32
> 
> batch_size=4, 8 GPU ecpoch=12 ，step=20292,train data num=50000

![image](https://user-images.githubusercontent.com/7944239/225518295-7431099d-a6ab-44d1-a8a2-826bf537434c.png)
this loss is ok ?@lllyasviel

**Comment by LiuXinqi12** [comment]
Hi， Have you solved this problem?

**Comment by shravankumar147** [comment]
How you managed to run on multiple gpus @qingfengmingyue ?

**Comment by HassanBinHaroon** [comment]
> 

By steps, Do you mean global steps (that is printed in the log)?

**Comment by kings-rgb** [comment]
Hi， Have you solved this problem?

**Comment by iLori-Jiang** [comment]
Also curious to know if you have solved this problem or not :)

**Comment by Neltherion** [comment]
Also curious!
